commit b49788abfe0d8c666b9b55c50b0849aab5d27792
Author: Timothy Potter <thelabdude@apache.org>
Date:   Tue May 20 13:20:28 2014 +0000

    SOLR-5468: Add wait loop to see replicas become active after restoring partitions; to address intermittent Jenkins test failures.
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/dev/trunk@1596234 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java b/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java
index 359568f..df422dd 100644
--- a/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java
+++ b/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java
@@ -234,6 +234,8 @@ public class ReplicationFactorTest extends AbstractFullDistribZkTestBase {
     getProxyForReplica(replicas.get(0)).reopen();    
     getProxyForReplica(replicas.get(1)).reopen();
     
+    ensureAllReplicasAreActive(testCollectionName, shardId, numShards, replicationFactor, 30);
+    
     rf = sendDoc(4, minRf);
     assertTrue("Expected rf=3 as partitions to replicas have been healed, but got "+rf, rf == 3);
     
